sdks/java/extensions/sql/datacatalog/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/DataCatalogBigQueryIT.java

```diff
@@ -66,12 +66,15 @@
           });
     }
 
+    @SuppressWarnings("initialization.fields.uninitialized")
     @Parameterized.Parameter(0)
     public String dialectName;
 
+    @SuppressWarnings("initialization.fields.uninitialized")
     @Parameterized.Parameter(1)
     public Class<? extends QueryPlanner> queryPlanner;
 
+    @SuppressWarnings("nullness")
```

Comment:
I would prefer not to suppress the warnings, but to fix them. Or if they are suppressed, we need a comment explaining why. I know this is just test code, but good test code matters too!

Reply:
I got this after enabling the checkerframework
```
/Users/jayendrap/lab/beam/sdks/java/extensions/sql/datacatalog/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/DataCatalogBigQueryIT.java:100: error: [argument.type.incompatible] incompatible types in argument.
          .setPlannerName(queryPlanner.getCanonicalName());
                                                       ^
  found   : @Initialized @Nullable String
  required: @Initialized @NonNull String
```
I am assuming that it is thrown as `queryPlanner` is uninitialized so compiler is inferring that `queryPlanner.getCanonicalName()` would give `null` and hence I suppressed that. Am I correct here, or I can't understand any other reason for this nullness.

Reply:
The checkerframework nullness analysis is not a heuristic warning framework like spotbugs/findbugs. It is a new and improved type system, so the idea is to always satisfy it by the right types everywhere.

In this case, the [javadoc for Class#getCanonicalName](https://github.com/typetools/jdk/blob/af987068c65767e101a3c58919cbae15ad08218b/src/java.base/share/classes/java/lang/Class.java#L1648) says "the canonical name of the class if it exists, otherwise null". So the checkerframework team has put a `@Nullable` annotation on the method. I actually did not realize this method was nullable. That is pretty annoying. So we might want to switch to just `getName` sometimes or otherwise use `MoreObjects.firstNonNull` to give a default string.

Reply:
gotcha so I am doing `queryPlanner.getCanonicalName()` --> `queryPlanner.getName()`.

Reply:
Maybe... in this case I realize that the string is probably used to create a planner by reflection. I don't know if that will work. If it does, then that is great. If it does not work, then you'll want to check for null and throw an exception. I've created a new Beam module to make this easy: `org.apache.beam.sdk.utils.Preconditions.checkArgumentNotNull` returns a non-nullable value after checking.

Reply:
Using `checkArgumentNotNull `
```
      readPipeline
          .getOptions()
          .as(BeamSqlPipelineOptions.class)
          .setPlannerName(checkArgumentNotNull(queryPlanner.getCanonicalName()));
```
throws 
```
/home/jayendra/beam/sdks/java/extensions/sql/datacatalog/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/DataCatalogBigQueryIT.java:101: error: [flowexpr.parse.error.postcondition] error parsing the postcondition expression for checkArgumentNotNull(T)
          .setPlannerName(checkArgumentNotNull(queryPlanner.getCanonicalName()));
                                              ^
  cannot parse the expression Invalid 'reference' because identifier not found
```
Which seems to be related to https://github.com/typetools/checker-framework/issues/752 .
I checked the other places in the code where `checkArgumentNotNull` has been used, all the uses are under the `runner.dataflow` package where null-cheking has not been applied yet.

Reply:
Yes, my mistake. I used the name `reference` when I needed to use the positional parameter `#1`. It should be fixed now.

Reply:
So should I put back `@SuppressWarnings("nullness")` for the method, but that would be method level suppression. What better we can do here ?

Reply:
I think I should mark this as `Resolved`.